### PR TITLE
Fix 64 to 32 bit clang conversion warning

### PR DESCRIPTION
**Pull request by dennycd**
Explicit casting to avoid 64 to 32 bit conversion clang warning ([-Wshorten-64-to-32](https://clang.llvm.org/docs/DiagnosticsReference.html#wshorten-64-to-32))  as seen from gRPC iOS and Firebase iOS SDK (e.g. https://github.com/firebase/firebase-ios-sdk/issues/9790, https://github.com/firebase/firebase-ios-sdk/discussions/9700) . This is observed from MacOS/iOS platform but I believe it also applies to others. thanks 

----

https://github.com/grpc/grpc-ios/issues/83

### CODE REVIEW

File: src/core/lib/security/credentials/external/aws_request_signer.cc [hunk 1]
```diff
@@ -17,8 +17,40 @@
...
+  #include <openssl_grpc/hmac.h>
+#else
+  #if COCOAPODS==1
+  #include <openssl_grpc/hmac.h>
+#else
+  #if COCOAPODS==1
+  #include <openssl_grpc/hmac.h>
+#else
+  #if COCOAPODS==1
+  #include <openssl_grpc/hmac.h>
```
Comment (sampajano) [inline src/core/lib/security/credentials/external/aws_request_signer.cc:30]: naive question - why are these look identical/repetitive? 😄

Comment (dennycd) [reply src/core/lib/security/credentials/external/aws_request_signer.cc]: updated and removed. These are local cocoapod changes that are not part of the PR

Comment (sampajano) [reply src/core/lib/security/credentials/external/aws_request_signer.cc]: ahh i see! makes more sense now! 😄

**Review by sampajano — APPROVED** [review]
thanks! :)

**Review by markdroth — CHANGES_REQUESTED** [review]
This doesn't look like the right way to fix this.  First, we should not be using C-style casts in C++ code.  And second, I think that instead of just adding casts to make the compiler problems go away, we should be looking at each of these cases to see what's actually causing the problem.

Just as one example, in the fault injection filter, instead of casting the result of `grpc_channel_stack_filter_instance_number()` to an int, we should change the type of the `index_` field in the filter from `int` to `size_t`.

**Comment by dennycd** [comment]
thanks Mark. will split this into individual case fixes

**Pull request closed without merging**